### PR TITLE
Update QSTN.md

### PR DESCRIPTION
**Pull request by qstnus**
#### Project Abstract

> QSTN is a Web3 survey marketplace where businesses fund surveys, reward participants, and uphold user privacy through our data wallet-driven profiles.

#### Grant [level] 1

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ Polkadot (USDC & USDT) or BTC address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by qstnus** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by keeganquigley** [comment]
For reference, see here https://github.com/w3f/Grant-Milestone-Delivery/pull/1017#issuecomment-1747270967

**Comment by keeganquigley** [comment]
Thanks for the update @qstnus so just to make sure I understand correctly:

- You are extending the timeline 3 months and aiming to deliver by end of December?
- Is all the rest of the info staying the same? For example, you [mentioned](https://github.com/w3f/Grants-Program/pull/1082#issuecomment-1739129336) the switch to traditional email OAuth instead of login with Polkadot SS58 accounts. Are you intending to implement functionality with wallets this time around?
- Seeing as it's been a year, are all team members the same?
- Is the legal entity and address still the same? I noticed that the application says QSTN, LLC while your website says @2023 Sink

**Comment by qstnus** [comment]
Hello @keeganquigley,

We are keeping the same proposal as outlined in our original application. Below are the answers to your question!

1. Yes we are extending until December (we should be done before this)

2. The information is staying the same - OAuth works on Moonbeam because we use Safe Global for the vault mechanism (EVM compatible), with Polkadot users need Polkadot JS wallet to interact with the pallet and custody funds.

3. The members are the same 

4. The legal entity and address is the same (Sink is the name of our lead developer's contracting firm)

**Comment by keeganquigley** [comment]
Thanks for the quick response @qstnus everything looks good except for regarding your second point:

- I understand the SAFE vault was needed for the EVM implementation "due to compatibility", but wouldn't this not be needed anymore for the native substrate version? Also, is the SAFE vault still non-custodial?

**Comment by qstnus** [comment]
Hello @keeganquigley 

SAFE is a non-custodial solution by Consensys (https://safe.global/), it facilitates wallet creation utilizing email addresses. For redeeming Polkadot rewards, users must link their JS wallets and claim rewards directly to their wallets.

Currently, SAFE lacks support for Polkadot. Given this, we will enable users to interact directly with the pallet, easing the reward redemption process post-survey completion.

EVM Scenario:
Upon signing up using an email address, EVM rewards are transferred to the user's SAFE vault.

Substrate Scenario:
Users sign up with an email address and connect their wallet, which allows for the withdrawal of DOT rewards.

**Comment by keeganquigley** [comment]
Thanks @qstnus for the updates. I will go ahead and mark the amendment as ready for review and ping the rest of the committee. Personally, I'm not willing to support it at this point as we've already waited a year for it, and I'm still not seeing the requested changes to the application. I'm not convinced about the technical aspects of it or whether it would be delivered in good faith, since Moonbeam is your team's main focus right now.

Also, since you already seem to have an established revenue model via [subscriptions](https://qstnus.com/), whereas we mostly focus on early-stage PoCs providing common good software for the community.

**Comment by qstnus** [comment]
Hello @keeganquigley, 

I appreciate your feedback. I wish to clarify that various restructurings, both within our product and team dynamics, significantly contributed to the delay in the grant's delivery.

Here are some notable milestones we've achieved since our grant request:

1. Secured a position in Outlier Venture's NEAR Base Camp, successfully closing our pre-seed round.
2. Developed a data wallet feature, now generating zero-knowledge proofs upon survey completion.
3. Integrated AI to assist businesses in crafting surveys and evaluating authenticity of responses through a "credibility" score.
4. Garnered preliminary interest from notable entities like NYU, Estée Lauder, and L'Oréal for potential test partnerships, in collaboration with the NEAR Foundation.
5. Were honored with acceptance into the Polkadot Blockchain Academy, although a visit was precluded due to visa constraints.

Press: 
Outlier Ventures - 
https://outlierventures.io/article/near-foundation-and-outlier-ventures-announce-the-first-near-base-camp-accelerator-cohort/ 

Parsiq -
https://hub.parsiq.net/unicornization 

Mastercard Demo Day -
https://youtu.be/TkfHLgsBxiE 

Mintbase -  
https://blog.mintbase.xyz/wrapping-up-mintbases-demo-day-f13fd192c8fc 

CoinDesk - 
https://www.coindesk.com/video/crypto-outlook-2023-regulation-web3-daos-stablecoins-and-jobs/

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Thanks @qstnus but again, these restructurings and findings would need to be integrated into the application itself. Along with the addition of the SAFE vault and any other changes. We understand that projects get delayed, but no amendments were ever filed to document these changes, despite me asking multiple times.

**Comment by qstnus** [comment]
Hello @keeganquigley,

How do you suggest we proceed? I am willing to add these revisions or should we leave as is? Thanks!

**Comment by qstnus** [comment]
Hello @keeganquigley,

In addition, our original application says 15 months, which puts the due date at October. I believe we can deliver within this time period and keep the same proposal if that would be an easier solution. Thanks!

**Comment by keeganquigley** [comment]
Thanks @qstnus feel free to make the changes, as my vote is just one out of the entire committee and other members may feel differently.

**Review by qstnus — COMMENTED** [review]
Improved application with new functionality, accomplishments and updated roadmap

**Review by qstnus — COMMENTED** [review]
I changed the deliverables to reflect the new functionality

**Comment by qstnus** [comment]
Hello @keeganquigley,

I appreciate the transparency. The application has been updated. Let me know if I need to do anything else!

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. What kind of ZK proofs will you use, and what are your long-term plans regarding Polkadot? To be honest, from your reply here, it sounds like we are supporting a NEAR project now.

File: applications/QSTN.md [hunk 1]
```diff
@@ -8,62 +8,90 @@
...
+Melanie Zhang, Creative Director: An NYU graduate with a keen eye for design and user experience, Melanie's artistic vision enriches the platform's aesthetics and fosters engagement with QSTN's users.
+
+Daniel Fadely, Financial Analyst: An NYU alum with a strong finance and investment background, Daniel's involvement in opening and running the investment club at NYU, along with experience in managing various portfolios, enhances the financial planning and analysis for QSTN.
 
 ### Project Details
 
 ### Ecosystem Fit
 
-- QSTN is Web3 infrastructure to accelerate the adoption of data monetization. Since users can earn with or without a Web3 wallet, but have the option to create one, shows we are designed to be a Web2 on-ramp into the world of Web3.
+- QSTN is an innovative Web3 survey platform that combines blockchain, gamification, and AI to revolutionize data collection. Businesses can easily fund surveys using digital tokens or NFTs, offering users enticing rewards for their participation. Our proprietary data wallet, Cubby, ensures users maintain full control over their data, ensuring privacy and security.
```
Comment (takahser) [inline applications/QSTN.md:50]: > Our proprietary data wallet, Cubby, ensures users maintain full control over their data, ensuring privacy and security.
 
This goes against our [guidelines](https://github.com/w3f/Grants-Program/tree/35fdc7317411e3cfc62cf9c7f1d53a9a962b55f1#guidelines):

> All code produced as part of a grant must be open-sourced, and it must also not rely on closed-source software for full functionality.

File: applications/QSTN.md [hunk 2]
```diff
@@ -8,62 +8,90 @@
...
 
-Anibal Suriel is Jose’s twin brother and our head of legal, he is registered in New York and Miami; leveraging his time at KARM to create the legal framework and guidelines around data, compliance, and overall best practices.
+Eduardo Gomes, Junior Developer: A promising talent, Eduardo contributes to QSTN's development efforts. With his skills and passion for identity, he adds value to the KYC and verification of users within the QSTN platform.
 
-With a desire to build meaningful products and experiences, Orrin founded QSTN with the aim to empower internet users and provide alternative means of passive income, aided by the invaluable expertise of the notorious Suriel duo.
+Melanie Zhang, Creative Director: An NYU graduate with a keen eye for design and user experience, Melanie's artistic vision enriches the platform's aesthetics and fosters engagement with QSTN's users.
+
+Daniel Fadely, Financial Analyst: An NYU alum with a strong finance and investment background, Daniel's involvement in opening and running the investment club at NYU, along with experience in managing various portfolios, enhances the financial planning and analysis for QSTN.
```
Comment (takahser) [inline applications/QSTN.md:44]: This seems to belong to the team section.

File: applications/QSTN.md [hunk 3]
```diff
@@ -204,29 +252,28 @@ Article 3: https://news.bloomberglaw.com/litigation/google-plus-7-5-million-priv
...
-| 2. | Survey pallet UI | We will create a new UI for DOT users to be able to ask questions, receive credit and screenshot their response on-chain; these design assets will include landing pages and layouts which we will open source to the DOT community 
-| 3. | Substrate chain | The survey pallet of our custom chain will interact in such a way to allow users to connect their DOT wallet to our application, mint media on a local substrate chain as well as create surveys, invite businesses to create their own surveys, and give permissions as to who can answer a survey [1.] and then save the completed survey ID on-chain as proof of completion [1.] these transactions take place within our new UI [2.]
+| 1. | Survey pallet | We will create a Substrate module that will allow users to complete a survey, earn a zero knowledge proof upon completion and then connect their Polkadot JS wallet in order to withdraw the pre-defined reward amount by interacting with the pallet.
```
Comment (takahser) [inline applications/QSTN.md:255]: > earn a zero knowledge proof

~~What does that mean? A proof for what? And what is it used for?~~

**EDIT:**

- Sorry, I just saw that you describe it in deliverable 5. I'll comment there directly.

File: applications/QSTN.md [hunk 4]
```diff
@@ -8,62 +8,90 @@
...
+
+To ensure this vision becomes a reality, we're harnessing the power of Safe's decentralized custody protocol and collective asset management platform. Users signing up on QSTN have their email addresses utilized to create individualized Safe vaults, powered by Web3 authentication. This integration enables users to complete surveys, earn zero-knowledge proofs, and subsequently claim their rewards directly within their EVM wallet. 
 
-We aim to do this by taking advantage of the low gas, scalable environment offered on DOT to allow users to earn credits, mint NFTs and transfer assets without exorbitant transactional fees. 
+SAFE is a non-custodial solution by Consensys (https://safe.global/), it facilitates wallet creation utilizing email addresses. Currently, SAFE lacks support for Polkadot. Given this, we will enable users to interact directly with the pallet, easing the reward redemption process post-survey completion. The wallet we will use is Polkadot JS wallet.
```
Comment (takahser) [inline applications/QSTN.md:54]: If I understood correctly, you're planning to build an integration between SAFE and Substrate, is that correct? Could you talk a bit more about how the integration would look like? Also, could you add it to the deliverables?

File: applications/QSTN.md [hunk 5]
```diff
@@ -204,29 +252,28 @@ Article 3: https://news.bloomberglaw.com/litigation/google-plus-7-5-million-priv
...
+| 2. | Survey pallet | We will create a Substrate module that will allow business to create a survey and then connect their Polkadot JS wallet in order to fund the pallet for respondents. 
+| 3. | Survey pallet UI | We will create a new UI for DOT users to be able to connect their Polkadot JS wallet, interact with the pallet to withdraw or fund survey rewards and mint their respective NFT media onto a local substrate chain. 
+| 4. | Substrate chain | The survey pallet of our custom chain will interact in such a way to allow users to connect their DOT wallet to our application, withdraw DOT and then mint media on a local substrate chain [1.] and these transactions take place within our new UI [3.] with proof recorded by our ZKP [5] after being funded by the business [2] 
+| 5. | ZKP Proof Generation | We will create an API for the Cubby data wallet to generate zero knowledge proofs upon survey completion on the Substrate chain - this ZKP will be the "receipt" to verify users are eligible to connect their wallet and withdraw rewards from a particular survey contract.
```
Comment (takahser) [inline applications/QSTN.md:259]: What's the benefit of using a ZKP here? It seems like you're just solving the problem of which users are eligible for rewards, right? Why not just write this information into storage?

**Comment by qstnus** [comment]
Hello Polkadot,

For clarification, we filed for this grant a year ago and decided to pivot our implementation after multiple feedback. I apologize for the confusion but DOT has always been in our roadmap. We started with Moonbeam first given its EVM compatibility and SAFE integration - now that is complete we will focus on the local substrate chain. 

It's a bit frustrating to have a grant potentially rescinded when we budgeted for this, hired developers specifically within the ecosystem from the PBA and it almost feels as if we are being punished for being a multichain application.

The reason DOT is attractive is because it aims to be intraoperative. Yes, we started this implementation on NEAR but given their connection with the Rust language - this is a natural evolution to our product. 

I have been following Polkadot since before the bull run and to work with the Web3 Foundation is a literal goal as a builder in the ecosystem for myself and our CTO. I have spent the past two years working very hard on this project as a solo and underrepresented founder so whatever information you need, I will procure. 

@Noc2 

Comment 1:
We are sourcing the ZK proof mechanism from Polygon ID (https://polygon.technology/polygon-id) - these cryptographic proofs occur off-chain and allow us to create attestations within the QSTN user profile after KYC and survey completion. 

In regards to future plans on Polkadot:
1. We plan to attend the PBA 5 Cohort in Singapore (spoken to Tim Dobie) and were confirmed for the last cohort but were unable to attend due to visa issues
2. We plan to integrate and partner with a marketplace on Polkadot to allow users to mint their survey media and access secondary trading liquidity 
3. We will apply for the Polkadot Global Relayers Program again - denied last time but will continue to apply 

Improvements from NEAR MVP to DOT:
1. Create leaderboards to track most active businesses and users 
2. Create credibility score using AI to filter user responses 
3. Create user funnel to allow business to index DOT customer profiles based on responses

@takahser 

Comment: 1
I removed the proprietary language from our documentation - I wrote it this way because we are patent pending for a couple methods and wanted to share our competitive moat but deliverable 5 is confirmation we will open source the ZKP API

Comment: 2
This has been moved to the correct section.

Comment: 3
We will not be building an integration between SAFE and Substrate because SAFE is only supporting EVM chains at the moment. I was explaining SAFE is the vault mechanism used in our EVM scenario but on DOT users have to claim with their Polkadot JS wallet directly. 

Comment: 4
A zero knowledge proof allows us to keep these attestations for KYC and/or survey completion within the user profile. Instead of writing in storage, we store these attestations in their user profile which doubles as a "data wallet" - in this manner, we allow users to autonomously withdraw rewards and our oracle scans for the ZKP.

If we act as an intermediary and make the reward determination, this is where we get into trouble as a US company given the regulatory uncertainty around securities, blockchains and our inability to facilitate user interactions with these assets.

**Comment by takahser** [comment]
@qstnus thanks for the updates and the detailed replies, I appreciate the work you're putting into this.

> We will create an API for the Cubby data wallet 

This means that the ZKPs are integrated with a proprietary product (Cubby) which still goes against our [guidelines](https://github.com/w3f/Grants-Program/tree/35fdc7317411e3cfc62cf9c7f1d53a9a962b55f1#guidelines), even if the deliverable itself is open-sourced:

> All code produced as part of a grant must be open-sourced, and it **must also not rely on closed-source software** for full functionality.

I'm fine with everything else, so if you would open-source Cubby (or maybe a downsized version thereof) I'd be willing to add my approval to the amendment. If that's not possible from a business perspective since you describe it as your _competitive moat_ I'd understand of course, but in that case I wouldn't be able to add my approval.

**Review by qstnus — COMMENTED** [review]
We added the data wallet as a deliverable

**Comment by qstnus** [comment]
Hello @takahser,

Thanks for the feedback. I spoke with our CTO and we will open source the data wallet as a deliverable. This has been included in our proposal. Please let me know if we should provide additional information. 

Looking forward to the next steps!

**Review by takahser — APPROVED** [review]
@qstnus thanks for the update. LGTM.

**Review by nikw3f — APPROVED** [review]

**Comment by nikw3f** [comment]
LGTM

**Comment by qstnus** [comment]
Hello @takahser & @nikw3f, 

Thanks for the support. We look forward to reaching mainnet on Polkadot. LGTM!

**Pull request merged**